src/interactive-window/interactiveWindow.ts

```diff
@@ -160,6 +160,28 @@ export class InteractiveWindow implements IInteractiveWindowLoadable {
         } else if (this.isWebExtension) {
             this.insertInfoMessage(DataScience.noKernelsSpecifyRemote()).ignoreErrors();
         }
+
+        this.listenForNewNotebookCells();
+    }
+
+    private listenForNewNotebookCells() {
+        workspace.onDidChangeNotebookDocument(
+            (e) => {
+                e.contentChanges.forEach((change) => {
```

Comment:
You'd need to filter this and ensure we only check for IW notebook types.

Reply:
Also, I think you can compare the notebook here, instead of doing that further below.
Else when you update a cell of another IW, we still end up running the for loop unnecessarily.